server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -312,10 +312,16 @@ public boolean isOpen() {
 
     @Override
     public void close() throws IOException {
+        close(true);
+    }
+
+    private void close(boolean force) throws IOException {
```

Comment:
Question - why is this needed?

Reply:
if we've got ioexception in `TranslogWriter#assertNoSeqAbove` on line `TranslogSnapshot snapshot = newSnapshot();` - TranslogWriter closes channel it by its own - after that `Translog#close` tries to perform `current.sync();` and fail on `TranslogWriter#ensureOpen` while other file handlers are not closed => therefore it's a way to avoid `current.sync()` is being called

Reply:
agreed to revert back to `seenSequenceNumbers` as it contains serialized form of Operation and able to perform primaryTerm check